### DISCUSSION TITLE
How to add external python libraries as dependencies inside my Rust application? I am getting error while trying to call a function from a python file that uses numpy and scikit-learn

### DISCUSSION THREAD

**Discussion by priyas2024** [PyO3/pyo3#4419] [General]
Hello all,
Inside  my rust application I am trying to call a simple function that runs a linear regression model and returns the predictions. This function uses the scikit-learn and numpy functions. I am trying to add the dependencies inside the pyproject.toml and use the maturin build. While running the rust function where the function call is made I am getting the NoModuleError for numpy.

Error: PyErr { type: <class 'ModuleNotFoundError'>, value: ModuleNotFoundError("No module named 'numpy'"), traceback: Some(<traceback object at 0x000001FD12E5D840>) }
error: process didn't exit successfully: `target\debug\rust_python3.exe` (exit code: 1)

This is my pyproject.toml
```
[build-system]
requires = ["maturin>=1,<2"]
build-backend = "maturin"

[project]
name = "rust_python"
version = "0.1.0"
dependencies = [
    "numpy",
    "scikit-learn == 1.1.3",
]
```
I am following the Pyo3 user guide example where we call an existing function inside a python file.

```
let from_python = Python::with_gil(|py| -> PyResult<Py<PyAny>> {
        PyModule::from_code_bound(py, py_ml, "utils.mlcheck","utils.mlcheck")?;
        let app: Py<PyAny> = PyModule::from_code_bound(py, py_app, "", "")?
            .getattr("run")?
            .into();
    //let args=(x,y);
        app.call0(py)
    });
```

I am following the below steps to run the application
1) pip install . 
to install all the dependencies
2) cargo build
to build the application 
3) cargo run
to run the rust application's  main.rs from where the call to the python function is made
I don't see any compile time errors. And the build completes without any issues.
Any help would be very much appreciated.

**Comment by davidhewitt** [comment]
What OS are you using? Might be a duplicate of #1741
